File: C++/Solution24.cpp
#include<iostream>
#include<vector>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution24 {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* pre = new ListNode(0, head);
        ListNode* ptr2 = pre;
        while(ptr2){
            ListNode* ptr = ptr2;
            ptr = ptr -> next;
            if(!ptr){
                return pre -> next;
            }
            ptr = ptr -> next;
            if(!ptr){
                return pre -> next;
            }
            ListNode* copy = ptr -> next;
            ptr2 -> next -> next = nullptr;
            ptr -> next = ptr2 -> next;
            ptr2 -> next -> next = copy;
            ptr2 -> next = ptr;
            ptr2 = ptr2 -> next -> next;
        }
        return pre -> next;
    }
};

File: Java/list/Solution24.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution24 {
    public ListNode swapPairs(ListNode head) {
    	if(head == null || head.next == null){
    		return head;
    	}

    	int temp = head.val;
    	head.val = head.next.val;
    	head.next.val = temp;
    	head.next.next = swapPairs(head.next.next);
    	return head;
    }
}